src/backend/distributed/planner/distributed_planner.c

```diff
@@ -1744,6 +1743,8 @@ multi_join_restriction_hook(PlannerInfo *root,
 	 */
 	plannerRestrictionContext->hasSemiJoin = plannerRestrictionContext->hasSemiJoin ||
 											 extra->sjinfo->jointype == JOIN_SEMI;
+	plannerRestrictionContext->hasOnlyInnerJoin = plannerRestrictionContext->hasOnlyInnerJoin &&
```

Comment:
I think we should move both hasOnlyInnerJoin and hasSemiJoin  to `JoinRestrictionContext`. With  that, we don't have to export/call `CurrentPlannerRestrictionContext`. (And, it has many implications anyway, we cannot do it).

Reply:
yeah, I added that to TODO's, will open a seperate pr for that